test/evp_kdf_test.c

```diff
@@ -295,12 +295,18 @@ static int do_kdf_hkdf_gettables(int extract_only, int has_digest)
             goto err;
     }
 
+    params_get[0] = OSSL_PARAM_construct_int(OSSL_KDF_PARAM_MODE, &mode_int);
+    params_get[1] = OSSL_PARAM_construct_end();
+    if (!TEST_int_eq(EVP_KDF_CTX_get_params(kctx, params_get), 1)
+        || !TEST_int_eq(mode_int, extract_only ? EVP_KDF_HKDF_MODE_EXTRACT_ONLY :
+                        EVP_KDF_HKDF_MODE_EXTRACT_AND_EXPAND))
+        goto err;
+
     params_get[0] = OSSL_PARAM_construct_utf8_string(OSSL_KDF_PARAM_MODE, mode_utf8,
                                                      sizeof(mode_utf8));
-    params_get[1] = OSSL_PARAM_construct_int(OSSL_KDF_PARAM_MODE, &mode_int);
-    params_get[2] = OSSL_PARAM_construct_octet_string(OSSL_KDF_PARAM_SALT, salt, sizeof(salt));
-    params_get[3] = OSSL_PARAM_construct_octet_string(OSSL_KDF_PARAM_INFO, info, sizeof(info));
-    params_get[4] = OSSL_PARAM_construct_end();
+    params_get[1] = OSSL_PARAM_construct_octet_string(OSSL_KDF_PARAM_SALT, salt, sizeof(salt));
+    params_get[2] = OSSL_PARAM_construct_octet_string(OSSL_KDF_PARAM_INFO, info, sizeof(info));
+    params_get[3] = OSSL_PARAM_construct_end();
```

Comment:
Hmm... but this looks like an API break. I know it is a very bad corner case but...

Reply:
But we have test cases with both variants of `extract_only` set and unset??

Actually it is an API break, because the original code does:

```
    p = params;
    while ((p = OSSL_PARAM_locate(p, OSSL_KDF_PARAM_MODE)) != NULL) {
        if (p->data_type == OSSL_PARAM_UTF8_STRING) {
        ....
        p++;
    }
```
I.e., it cycles through all OSSL_KDF_PARAM_MODE in the array.

Reply:
That code isn't in 3.5 that I can see, so it's not been released.  It was added by #27365 which was master only.
So still not an API break.

Nonetheless, it's easy to address and handle multiple copies of the param.

Reply:
I can do it here or as a follow up PR.  I'd prefer the latter in order to expedite this.

Reply:
> It was added by #27365 which was master only.
> So still not an API break.

Ahh.... then we are good, I think. I am not even sure it makes sense to support such corner use-case. If anything the caller can always call get params twice.

Reply:
A committer or TAC discussion about the sensibility of having it?

It wouldn't be hard to implement but it should be done after some thought and if it is done it must be done everywhere this possibility arises.

Reply:
I've raised the decision about which way to process these in #28099.